File: leetcode/algorithms/c++/Number of Steps to Reduce a Number to Zero/Number of Steps to Reduce a Number to Zero.cpp
class Solution
{
public:
    static inline int numberOfSteps(int num)
    {
        if (num == 0)
            return 0;
        if (num == 1)
            return 1;
        if (num % 2 == 0)
            return numberOfSteps(num / 2) + 1;
        return numberOfSteps(num - 1) + 1;
    }
};

File: leetcode/algorithms/c#/Number of Steps to Reduce a Number to Zero/Number of Steps to Reduce a Number to Zero.cs
public class Solution
{
    public int NumberOfSteps(int num)
    {
        if (num == 0) return 0;
        if (num == 1) return 1;
        if (num % 2 == 0) return NumberOfSteps(num / 2) + 1;
        return NumberOfSteps(num - 1) + 1;
    }
}
